test/specs/validate-address.spec.js

```diff
@@ -154,38 +253,278 @@ describe("validateAddress()", () => {
     assertNormalizedAddressFormat(normalizedAddress);
 
     // The normalized address should match the original address
-    assertNormalizedAddressMatchesOriginal(normalizedAddress);
+    assertNormalizedAddressMatchesOriginal(
+      addressToValidate,
+      normalizedAddress
+    );
+
+    // It should not throw errors
+    expect(response.info).to.be.an("array").and.be.empty;
+    expect(response.warnings).to.be.an("array").and.not.be.empty;
+    expect(response.errors).to.be.an("array").and.be.empty;
+    expect(response.warnings[0]).to.equal(
+      "This address has been verified down to the house/building level (highest possible accuracy with the provided data)"
+    );
+  });
+
+  it("Validates an address with errors", async function () {
+    const shipengine = new ShipEngine({ apiKey, baseURL });
+
+    const addressToValidate = {
+      country: "CA",
+      street: ["170 Princes' Blvd", "validate-with-error"],
+      cityLocality: "Toronto",
+      stateProvince: "On",
+      postalCode: "M6K 3C3",
+    };
+
+    const response = await shipengine.validateAddress(addressToValidate);
+
+    const { normalizedAddress, isValid } = response;
+
+    expect(isValid).to.be.a("boolean").and.to.be.false;
+    expect(normalizedAddress).to.be.undefined;
+
+    // It should not throw errors
+    expect(response.info).to.be.an("array").and.be.empty;
+    expect(response.warnings).to.be.an("array").and.be.empty;
+    expect(response.errors).to.be.an("array").and.not.be.empty;
+    expect(response.errors[0]).to.equal("Invalid City, State, or Zip");
+  });
+
+  it("Throws an error if no address lines are provided", async function () {
+    const shipengine = new ShipEngine({ apiKey, baseURL });
+    let error;
+    const addressToValidate = {
+      country: "US",
+      street: [],
+      cityLocality: "Boston",
+      stateProvince: "MA",
+      postalCode: "02215",
+    };
+
+    try {
+      await shipengine.validateAddress(addressToValidate);
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).to.be.not.undefined;
+    expect(error.source).to.equal("shipengine");
+    expect(error.type).to.equal("validation");
+    expect(error.code).to.equal("field_value_required");
+    expect(error.message).to.equal(
+      "Invalid address. At least one address line is required."
+    );
+    expect(error.requestId).to.be.undefined;
```

Comment:
Rather than just asserting that `error` is not `undefined`, assert that it's an instance of the expected error class.  For example:

```javascript
expect(error).to.be.an.instanceOf(ShipEngineError);
expect(error).to.be.an.instanceOf(FieldValueRequiredError);
```

**OR** you could just use the helper functions that I created in [`test/utils/errors.js`](https://github.com/ShipEngine/shipengine-js/blob/main/test/utils/errors.js) instead.  See [these config tests](https://github.com/ShipEngine/shipengine-js/blob/6b9d2ebe3f2e5c80a67e69041739c4533929e81b/test/specs/config.spec.js#L67-L96) for examples.  

```suggestion
    errors.assertFieldValueRequiredError(error, {
      source: "shipengine",
      type: "validation",
      code: "field_value_required",
      message: "Invalid address. At least one address line is required.",
      requestID: undefined,
    });
```

Reply:
Thanks! I'll definitely use your helper functions.